### PR TITLE
Re mk cross build2

### PR DESCRIPTION
**Pull request by cspiel1**
Allows a cross compilation build by only setting SYSROOT.

### CODE REVIEW

File: mk/re.mk [hunk 1]
```diff
@@ -725,6 +726,11 @@ rpm:    tar
 # - system installation
 #
 
+ifneq ($(SYSROOT),/usr)
+LIBRE_PATH := $(shell [ -f $(SYSROOT)/include/re/re.h ] && \
+	echo "$(SYSROOT)")
+endif
+
```
Comment (sreimers) [inline mk/re.mk:733]: I think we should guarantee the prioritised order for relative path re search,
Like this (untested):

```mk
ifeq ($(LIBRE_PATH),)
LIBRE_PATH := $(shell [ -d ../re ] && echo "../re")
endif

ifeq ($(LIBRE_PATH),)
LIBRE_PATH := $(shell [ -f $(SYSROOT)/include/re/re.h ] && \
	echo "$(SYSROOT)")
endif
```

Comment (sreimers) [reply mk/re.mk:733]: So the new order we should document:

# - relative path (../re)
# - custom sysroot
# - local installation (/usr/local)
# - system installation (/usr)

**Comment by cspiel1** [comment]
The prio order for re makes sense. I don't know why the retest failed for macOS. And I don't understand this build command:
```
cd retest && make STATIC=1 LIBRE_SO=../re && ./retest -r
```
STATIC is not used in retest.

**Comment by sreimers** [comment]
The build is error is unrelated. It looks like a race condition since the jbuf changes (occurs sometimes):

```
testjbuf: TEST_ERR: src/jbuf.c:234: (No such file or directory)
test: test_jbuf_adaptive: test failed (No such file or directory)
```
I had no time to investigate. 

> STATIC is not used in retest.

Good point, i will remove.

**Pull request merged**